### ISSUE TITLE
using cmake to compile in vs2019 + windows (64) + cuda10.1，Eigen function in kernel function of libvis_cuda lead to error

### ISSUE THREAD

**Issue by zhulz** [ETH3D/badslam#54]
use cmake to compile in vs2019 + windows (64) + cuda10.1, in libvis_ CUDA's Cu file there has the eigen function and define. this markes that  libvis_ CUDA is failed to compile. i find that you provide the exe files in the windows ,   How do you compile it in windows ?
@puzzlepaint wish your help!

**Comment by puzzlepaint** [comment]
It is possible that an Eigen include got introduced in a .cu file again after the Windows version was compiled. Which file is it? The include should be removed from that file to fix the problem.

**Comment by zhulz** [comment]
When I reconfigured and then compiled, it passed. Thanks

**Comment by zapaishchykova** [comment]
Hey, Im trying to compile on similar setup - vs19+cuda 10.2 + win 10 x64, and I get the following error:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at D:/Installed/VisualStudio2019/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Eigen3 (missing: EIGEN3_INCLUDE_DIR EIGEN3_VERSION_OK)
  (Required is at least version "2.91.0")		D:/Installed/VisualStudio2019/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake	164	
```
Can you please describe how did you reconfigure it? Many thanks!

**Comment by zhulz** [comment]
the error gives you good tips.  that is , your eigen3 library  is not  included

**Comment by zapaishchykova** [comment]
@zhulz so i tried to include this in the beginning of the Makefile
```
set(EIGEN3_INCLUDE_DIR "D:\\Installed\\eigen\\eigen-3.3.7\\eigen-3.3.7\\cmake")
include_directories(${EIGEN3_INCLUDE_DIR})
```
And it did not help, unfortunately :(

**Comment by zhulz** [comment]
1. When you compile the code on windows. What it needs is cmakelist.txt not makefile.
2. Eigen library is not needed to compile,  if the path of the eigen is  xxx\\xxx\\Eigen\\Dense., you try to set set(EIGEN3_INCLUDE_DIR "xxx\\xxx"), the Eigen library is used by including "#include<Eigen/Dense>" in projection or solution. so you only set the path xxxx\\xxxx.

**Comment by zapaishchykova** [comment]
Sorry my bad, i meant its in the cmakelist.txt. I did not understand what you mean by the 2nd point, can you clarify it?

**Comment by zhulz** [comment]
for  example ,  the path of eigen library is "‪E:\TOOLE\badslam111\badslam-master\extlib\Eigen337\Eigen\Dense" in my computer.
![2](https://user-images.githubusercontent.com/12679016/93009400-2cee7e80-f5b3-11ea-9373-8143f32349a0.jpg)


the cmake-gui is configured as below: 
![1](https://user-images.githubusercontent.com/12679016/93009387-07fa0b80-f5b3-11ea-9853-a73a0e4b4b77.jpg)

**Comment by zapaishchykova** [comment]
I see, I tried to set also one level higher and issue is still the same. Did you succeed compiling it on Win with Cmake?

**Comment by zhulz** [comment]
yes, i compiled it successfully.

**Comment by zhulz** [comment]
you can paste how you configure it in cmake-gui and the path of the library.  maybe i can  find some problems  and  give you some helps.

**Comment by zapaishchykova** [comment]
Thanks! Now I'm bothered by the boost configuration error:
```
CMake Error at D:/Installed/Cmake/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Boost (missing: serialization filesystem system) (found
  version "1.66.0")

```
Although I set it up D:\Download\boost_1_66_0
![image](https://user-images.githubusercontent.com/17569461/93009879-ff261d00-f553-11ea-935e-8f523f0eb012.png)
How to set up it correctly?

**Comment by zhulz** [comment]
maybe this picture is useful!
![3](https://user-images.githubusercontent.com/12679016/93009916-f61b6700-f5b8-11ea-8a77-52fe86edc27e.jpg)

**Comment by zapaishchykova** [comment]
My problem is that i don't have a vc142-x64 folder inside boost_1_73:( Did you compile/install boost library too?

**Comment by zhulz** [comment]
yes, i compile the boost library.  the boost library .The boost library can be compiled with vs17 or vs15. 

You can first refer to my configure figure and add the .lib at the corresponding position in cmake GUI. Although there is no v142-x64 library, other versions of the library may also work.

for example,  in the cmake-gui  listview , the  value of  "Boost_SERIALIZATION_LIBRARY_RELEASE "  also can be set as " boost_serialization-vc140-mt-x64-1_66.lib"  or other version. You can try it and see if it works.

**Comment by meitiever** [comment]
I got this error when compiling with win 10 + CUDA 11 + vs2019 community.
![WeChat Image_20210308003355](https://user-images.githubusercontent.com/4424258/110247200-ffa0ed00-7fa5-11eb-975c-240ddd2fec12.png)

**Comment by zhulz** [comment]
From the error information, QT is not configured well ， some lib  have not added！

**Comment by meitiever** [comment]
I have included all qt libs of 5.9.9 or 5.14.0.  And the error also pops up.
